apps/05Mastermind.js

```diff
@@ -9,67 +9,104 @@ var board = [];
 var solution = '';
 var letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h'];
 
+
 function printBoard() {
-    for (var i = 0; i < board.length; i++) {
-        console.log(board[i])
-    }
+  for (var i = 0; i < board.length; i++) {
+    console.log(board[i])
+  }
 }
 
 function generateSolution() {
-    for (var i = 0; i < 4; i++) {
-        var randomIndex = getRandomInt(0, letters.length);
-        solution += letters[randomIndex];
-    }
+  for (var i = 0; i < 4; i++) {
+    var randomIndex = getRandomInt(0, letters.length);
+    solution += letters[randomIndex];
+  }
 }
 
 function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-    // your code here
+function generateHint(solution, guess) {
+  var solutionArray = solution.split('');
+  var guessArray = guess.split('');
+  var correctLetterLocations = 0;
+
+
+  for (var i = 0; i < solutionArray.length; i++) {
+    if (solutionArray[i] === guessArray[i]) {
+      correctLetterLocations++;
+      solutionArray[i] = null;
+    }
+  }
+
+  var correctLetters = 0;
+
+  for (i = 0; i < solutionArray.length; i++) {
+    var targetIndex = solutionArray.indexOf(guessArray[i]);
+    if (targetIndex > -1) {
+      correctLetters++;
+      solutionArray[targetIndex] = null;
+    }
+  }
+
+  return correctLetterLocations.toString() + '-' + correctLetters.toString();
 }
 
+
+
 function mastermind(guess) {
-    // your code here
+
+  solution = 'abcd';
```

Comment:
This doesn't break the tests, but I bet it overrides running the app with `node`. We probably want to remove this so the game can use randomly generated solutions